docs/diagrams/ModelClassDiagram.puml

```diff
@@ -5,46 +5,37 @@ skinparam arrowColor MODEL_COLOR
 skinparam classBackgroundColor MODEL_COLOR
 
 Package Model <<Rectangle>>{
-Class "<<interface>>\nReadOnlyAddressBook" as ReadOnlyAddressBook
+Class "<<interface>>\nReadOnlyShop" as ReadOnlyShop
 Class "<<interface>>\nReadOnlyUserPrefs" as ReadOnlyUserPrefs
 Class "<<interface>>\nModel" as Model
-Class AddressBook
+Class Shop
 Class ModelManager
 Class UserPrefs
 
-Class UniquePersonList
-Class Person
-Class Address
-Class Email
-Class Name
-Class Phone
-Class Tag
-
+Class UniqueXList
+Class X
+Class PartMap
+'Class Part
 }
```

Comment:
Think you missed out `ServiceList which is also something `Shop` also has an association with, ServiceList is implemented similarly to `UniqueXList` but without the unique constraint within the list

Reply:
like dis?
![image](https://user-images.githubusercontent.com/89006592/227279381-5bbab0ca-a280-4268-ba83-d1dc3545b190.png)

Reply:
Y can just be Service(?)